core/shared/src/test/scala/fs2/concurrent/ChannelSuite.scala

```diff
@@ -251,4 +251,28 @@ class ChannelSuite extends Fs2Suite {
 
     TestControl.executeEmbed(test)
   }
+
+  test("sendAndClose closes right after sending the last element") {
+    val result = Channel.bounded[IO, Int](1).flatMap { ch =>
+      ch.sendAndClose(0) *> ch.isClosed
+    }
+
+    result.assertEquals(true)
+  }
+
+  test("racing send and sendAndClose should work in bounded case") {
+    val test = Channel.bounded[IO, Int](2).flatMap { ch =>
+      ch.send(0).both(ch.sendAndClose(1))
+    }
+    test.assertEquals((Right(()), Right(())))
```

Comment:
I could be wrong, but I think remark from @armanbilge (https://github.com/typelevel/fs2/pull/3280#discussion_r1301707331) makes a lot of sense, i.e. maybe we should cover both possible cases and assert which elements were actually sent in the Channel (for example the current assertion would pass even if `send` would still succeed *after* `sendAndClose`). :slightly_smiling_face: